src/main/flight/imu.c

```diff
@@ -708,20 +699,20 @@ void imuQuaternionMultiplication(quaternion *q1, quaternion *q2, quaternion *res
     result->z = D + (+ E - F - G + H) / 2.0f;
 }
 
-void imuQuaternionHeadfreeTransformVectorEarthToBody(t_fp_vector_def *v)
+void imuQuaternionHeadfreeTransformVectorEarthToBody(vector3_t *v)
 {
     quaternionProducts buffer;
```

Comment:
(quaternionProducts is overoptimization, and is probably making code slower - multiplication is single cycle, faster than memory access)

Reply:
I'm not sure what to do here, I just left the code as is. What can I do to replace this all with a single multiplication?

Reply:
It's not directly related to this PR, just note to existing code. Separate PR can remove quaternionProducts